lib/modules/dosomething/dosomething_northstar/dosomething_northstar.module

```diff
@@ -153,36 +155,35 @@ function dosomething_northstar_update_user($user, $password = null) {
 }
 
 /**
- * Get user profile data from Northstar for the given Drupal ID.
+ * Get user profile data from Northstar by the specified ID.
  *
- * @param int $drupal_id Drupal user id.
- * @return object
+ * @param  string $id
+ * @param  string $type
+ * @return DoSomething\Northstar\Resources\NorthstarUser|null
  */
-function dosomething_northstar_get_northstar_user($drupal_id) {
-  $northstar_user_info = cache_get('northstar_user_info_' . $drupal_id, 'cache_dosomething_northstar');
+function dosomething_northstar_get_user($id, $type = 'drupal_id') {
+  $cache_bin = 'cache_dosomething_northstar';
+  $cache_key = 'northstar_user_'.$type.'='.$id;
+
+  $northstar_user = cache_get($cache_key, $cache_bin);
 
-  if ($northstar_user_info) {
-    $northstar_user_info = $northstar_user_info->data;
+  if ($northstar_user) {
+    return $northstar_user->data;
```

Comment:
Does this work? I don't believe `data` is accessible unless you're somehow extracting the raw JSON.